cpp/ql/lib/semmle/code/cpp/dataflow/internal/DataFlowImplCommon.qll

```diff
@@ -80,8 +80,8 @@ predicate accessPathCostLimits(int apLimit, int tupleLimit) {
 pragma[noinline]
 predicate argumentPositionMatch(DataFlowCall call, ArgNode arg, ParameterPosition ppos) {
   exists(ArgumentPosition apos |
-    arg.argumentOf(call, apos) and
-    parameterMatch(ppos, apos)
+    arg.argumentOf(call, pragma[only_bind_into](apos)) and
```

Comment:
I would prefer to revert this change, and instead add `pragma[nomagic]` to the `parameterMatch` predicate.

Reply:
done. let's talk over later on why this is better :blush: